packages/example/src/components/modals/GenericModal.tsx

```diff
@@ -0,0 +1,47 @@
+import styled from '@emotion/native';
+import React from 'react';
+import { View, ModalProps } from 'react-native';
+import { Typography } from '../../design-system/components/Typography';
+import { Spacer } from '../../design-system/components/Spacer';
+
+type CustomModalProps = ModalProps & {
+  isVisible: boolean;
+  children: React.ReactNode;
+  title: string;
+};
+
+export const GenericModal = ({ isVisible, children, title }: CustomModalProps) => {
```

Comment:
This should contain the spatial navigation logic. You can call it SpatialNavigationModal maybe?